server/test/list_objects.go

```diff
@@ -187,7 +187,7 @@ func runListObjectsTests(t *testing.T, ctx context.Context, testCases []listObje
 			require.ErrorIs(t, err, test.expectedError)
 
 			if len(streamedObjectIds) > defaultListObjectsMaxResults {
-				t.Errorf("expected a maximum of %d results but got %d:", defaultListObjectsMaxResults, len(streamedObjectIds))
+				t.Errorf("expected a maximum of %d results but got %d", defaultListObjectsMaxResults, len(streamedObjectIds))
 			}
 			if !subset(streamedObjectIds, test.expectedResult) {
```

Comment:
Can we please use `require.Subset` here instead? No need to maintain the extra `subset` method here.